### PR TITLE
[doc] fix RCP building information for nRF52840

### PR DESCRIPTION
**Pull request by edmont**

### CODE REVIEW

File: src/posix/README.md [hunk 1]
```diff
@@ -77,19 +77,25 @@ To build and program the device with RCP application, complete the following ste
 
    ```sh
    rm -rf build
-   script/build nrf52840 USB_trans -DOT_BOOTLOADER=USB -DOT_THREAD_VERSION=1.2
+   script/build nrf52840 USB_trans -DOT_BOOTLOADER=USB -DOT_THREAD_VERSION=1.3
```
Comment (jwhui) [inline src/posix/README.md:80]: Can we simply remove the `-DOT_THREAD_VERSION=1.3`?

Comment (abtink) [reply src/posix/README.md:80]: +1 on this, using default version may be better.

Comment (edmont) [reply src/posix/README.md]: Makes sense.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**